src/migrations/20171017150457_market_state.ts

```diff
@@ -0,0 +1,18 @@
+import * as Knex from "knex";
+
+// PRIMARY KEY AUTOINCREMENT pgsql compat issue? TODO: use .primary()
```

Comment:
Please use the knex functions as needed to address compatibility issues.  Actually, there appear to be enough subtle differences between sqlite/pgsql that it might be sensible for us to solely use the knex wrappers unless we're doing an abnormally complicated query / really need the raw sql input capability.

Reply:
Did you see the above conversation about this? There's a bug in knex i'm working around; it is unable to add a key like this after table create.

Reply:
Ah!  Sorry about that, no I did not.  Is it possible to use knex to generate the create table command itself, and include the primary key autoincrement as part of that?

Reply:
I'll see about changing the whole thing to a knex-based create; seems like that might work and be the right solution anyway.